lib/integrations/awsra/generate_credentials.go

```diff
@@ -249,3 +252,30 @@ func GenerateCredentials(ctx context.Context, req GenerateCredentialsRequest) (*
 		SerialNumber:    x509Cert.SerialNumber.String(),
 	}, nil
 }
+
+// RolesAnywhere CreateSessionAPI does not mention any validation of the RoleSessionName field:
+// https://docs.aws.amazon.com/rolesanywhere/latest/userguide/authentication-create-session.html
+// However, the API returns the following error, if it doesn't match the expected format:
+// > Value '<xyz>' at 'roleSessionName' failed to satisfy constraint: Member must satisfy regular expression pattern: [a-zA-Z0-9_=,.@-]*"
+func roleSessionNameFromSubject(ctx context.Context, subject string) string {
+	roleSessionName := strings.Builder{}
+	changed := false
+	for _, r := range subject {
+		if (r >= 'a' && r <= 'z') || (r >= 'A' && r <= 'Z') || (r >= '0' && r <= '9') || slices.Contains([]rune{'_', '=', ',', '.', '@', '-'}, r) {
+			roleSessionName.WriteRune(r)
+		} else {
+			roleSessionName.WriteRune('_')
+			changed = true
+		}
+	}
+
+	if changed {
+		slog.InfoContext(ctx,
+			"AWS role session name does not comply with Roles Anywhere CreateSession API, replaced invalid chars with _.",
+			"original", subject,
+			"role_session_name", roleSessionName.String(),
+		)
+	}
+
+	return roleSessionName.String()
```

Comment:
suggestion: use a regexp replace

```suggestion
	unsupportedChars := regexp.MustCompile("[^a-zA-Z0-9_=,.@-]")
	updatedSubject := unsupportedChars.ReplaceAllString(subject, "_")

	if updatedSubject != subject {
		slog.InfoContext(ctx,
			"AWS role session name does not comply with Roles Anywhere CreateSession API, replaced invalid chars with _.",
			"original", subject,
			"role_session_name", updatedSubject,
		)
	}

	return updatedSubject
```